### PR TITLE
JSF itests stabilization

### PR DESCRIPTION
**Pull request by odockal**
- Workaround for infinite building workspace loop
   - New way of downloading jboss-eap 7.0 server, via download-maven-plugin
   - Itests are green on jenkins rhel and win slaves

Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

**Comment by odockal** [comment]
Awaits https://github.com/jbosstools/jbosstools-javaee/pull/451 to be merged in javaee repo.

**Comment by odockal** [comment]
Jenkins job: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/jsf.itests/160/

**Comment by odockal** [comment]
testPR

**Comment by odockal** [comment]
testPR

**Comment by jkopriva** [comment]
@psrna @rawagner can you please review?

**Comment by jkopriva** [comment]
@odockal just a side note: why don't we use eap 7.1?

**Comment by odockal** [comment]
@jkopriva No special reason. Can be updated imo.

**Comment by odockal** [comment]
Updated, jenkins build is being run. let us see the result...

**Comment by jkopriva** [comment]
@psrna @rawagner can you please review/merge?

**Comment by odockal** [comment]
@psrna @jkopriva updated to work with jboss-eap-7.1.0 by default. Jenkins build is green: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/Devstudio/view/devstudio_integration_tests/job/jsf.itests/

File: tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java [hunk 1]
```diff
@@ -50,9 +57,13 @@
 @DoNotUseVPE
 @UseParametersRunnerFactory(ParameterizedRequirementsRunnerFactory.class)
 @JBossServer(state = ServerRequirementState.RUNNING)
+@CloseAllEditors
 public class RunJSFProjectTest {
 
 	protected static final String PROJECT_NAME_BASE = "JSFTestProject";
+	
+	@SuppressWarnings("unused")
+	private static boolean IS_WINDOWS = System.getProperty("os.name").startsWith("Windows") ? true : false;
```
Comment (psrna) [inline tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: Where do you use this?

File: tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java [hunk 2]
```diff
@@ -114,15 +164,27 @@ private void checkDeployedProject() {
 	}
 
 	private void deployProject() {
+		new WaitUntil(new JobIsRunning(), TimePeriod.MEDIUM, false);
```
Comment (psrna) [inline tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: What do you wait for?

Comment (odockal) [reply tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: Have to wait for server adapter to get created, sometimes I got error here saying, there is no server adapter... but in stack trace you can see it. This seemed to solve the problem.

File: tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java (hunk 1 above)
Comment (odockal) [reply tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: I do not use it anymore, that is why is suppress warning annotation there... will remove.

File: tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java (hunk 2 above)
Comment (psrna) [reply tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: Perhaps this should be fixed in server requirement or here: https://github.com/eclipse/reddeer/blob/820e8f1891451870c0569173864b76f0286bd2c4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/wizard/NewServerWizard.java#L38

@rawagner any suggestions?

Comment (rawagner) [reply tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java]: We could add a wait which checks the TreeItem presence in the ServersView. Issue is here https://github.com/eclipse/reddeer/issues/1899

**Comment by odockal** [comment]
@psrna @rawagner I have run tests multiple times again and did not reproduce the issue again. I have removed wait.

**Comment by odockal** [comment]
@psrna jenkins is green.

**Comment by odockal** [comment]
@psrna @rawagner ready to merge? :)

**Pull request merged**